models/user.go

```diff
@@ -10,24 +10,62 @@ import (
 	"golang.org/x/crypto/bcrypt"
 )
 
+// User represents a user account.
 type User struct {
-	ID        int64  `json:"id,omitempty"`
-	Name      string `json:"name,omitempty"`
-	Email     string `json:"email,omitempty"`
-	Password  string `json:"password,omitempty"`
-	AvatarURL string `json:"avatar_url,omitempty"`
+	ID        int64   `json:"id,omitempty"`
+	Name      string  `json:"name,omitempty"`
+	Email     string  `json:"email,omitempty"`
+	Password  string  `json:"password,omitempty"`
+	AvatarURL *string `json:"avatar_url,omitempty"`
 }
 
 const (
 	usersTable string = "users"
 )
 
+// Merge creates a new User by copying the original User and replacing its
+// fields with the non-zero-value fields of a patch User.
+func (u *User) Merge(patch *User) *User {
+	newUser := &User{ID: u.ID}
+
+	if patch.Name != "" {
+		newUser.Name = patch.Name
+	} else {
+		newUser.Name = u.Name
+	}
+
+	if patch.Email != "" {
+		newUser.Email = patch.Email
+	} else {
+		newUser.Email = u.Email
+	}
+
+	if patch.Password != "" {
+		newUser.Password = patch.Password
+	} else {
+		newUser.Password = u.Password
+	}
+
+	if patch.AvatarURL != nil {
+		newUser.AvatarURL = patch.AvatarURL
+	} else {
+		newUser.AvatarURL = u.AvatarURL
+	}
+
+	return newUser
+}
+
 // CheckUser authenticates the user's credentials and returns the user.
 func (db *DB) CheckUser(user *User) (*User, error) {
 	userFromDB := &User{}
-	queryString := fmt.Sprintf("SELECT ID, Name, Password FROM %s WHERE EMAIL=?", usersTable)
-	err := db.QueryRow(queryString, user.Email).Scan(&(userFromDB.ID),
-		&(userFromDB.Name), &(userFromDB.Password))
+	queryString := fmt.Sprintf("SELECT ID, Name, Email, Password, AvatarUrl FROM %s WHERE EMAIL=?", usersTable)
```

Comment:
Is the idea now that `heimdall` is going to return these fields to the client when it hits the `/heimdall/v1/token` route, i.e. signs in?

Reply:
Um, no... I think I was just on autopilot and didn't think about it. I was just treating it like a regular `POST` essentially. Do you think the handler should just return `"id"`, `"name"`, and `"email"`, then?

Reply:
Oh haha yeah I think so because that's all `heimdall` actually needs.